src/app/copy/copy-service/copy.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Inject, Injectable } from '@angular/core';
+import { DOCUMENT } from '@angular/platform-browser';
```

Comment:
My IDE suggests DOCUMENT is deprecated? The docs say "import from `@angular/common` instead."

Reply:
I hadn't yet addressed this in that last commit I pushed, was getting all the familiar things out of the way first :)

I have it in place now and seems to be working just fine! Will be included in this next commit.